library/ssl_tls13_server.c

```diff
@@ -1749,6 +1749,25 @@ static int ssl_tls13_parse_client_hello(mbedtls_ssl_context *ssl,
     return hrr_required ? SSL_CLIENT_HELLO_HRR_REQUIRED : SSL_CLIENT_HELLO_OK;
 }
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+static void ssl_tls13_update_early_data_status(mbedtls_ssl_context *ssl)
+{
+    mbedtls_ssl_handshake_params *handshake = ssl->handshake;
+
+    if ((handshake->received_extensions &
+         MBEDTLS_SSL_EXT_MASK(EARLY_DATA)) == 0) {
+        MBEDTLS_SSL_DEBUG_MSG(
+            1, ("EarlyData: no early data extension received."));
```

Comment:
```suggestion
        MBEDTLS_SSL_DEBUG_MSG(
            2, ("EarlyData: no early data extension received."));
```
Non-Blocker: The absence of `early_data` extension is not kind of error nor something fatal. So I suggest to use level `2` or `3`.

It's not worth to iterate the PR, we could improve it in a follow-up.